ModelBundle/EventListener/TrashCanListener.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace OpenOrchestra\ModelBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\PostFlushEventArgs;
+use Doctrine\ODM\MongoDB\Event\PreUpdateEventArgs;
+use OpenOrchestra\ModelBundle\Document\TrashCan;
+use OpenOrchestra\ModelInterface\Model\TrashCanableInterface;
+use Symfony\Component\DependencyInjection\ContainerAware;
+use DateTime;
+
+/**
+ * Class TrashCanListener
+ */
+class TrashCanListener extends ContainerAware
+{
+    public $entities = array();
+
+    /**
+     * @param PreUpdateEventArgs $event
+     */
+    public function preUpdate(PreUpdateEventArgs $event)
+    {
+        $document = $event->getDocument();
+        if ($document instanceof TrashCanableInterface) {
+            if ($event->hasChangedField('deleted') && $event->getNewValue('deleted') === true && $event->getOldValue('deleted') === false) {
+                $trashCanClass = $this->container->getParameter('open_orchestra_model.document.trash_can.class');
+                /**
+                 * @var TrashCan $trashCan
+                 */
+                $trashCan = new $trashCanClass();
+                $trashCan->setEntity($document);
+                $trashCan->setName($document->getTrashCanName());
+                $date = new DateTime();
```

Comment:
The date should be set automatically in the __construct from the trashcan entity